third_party/move/move-compiler-v2/src/pipeline/ability_processor.rs

```diff
@@ -172,6 +172,7 @@ impl<'a> TransferFunctions for CopyDropAnalysis<'a> {
         let temp_needs_drop = |temp: &TempIndex| {
             self.target.get_local_type(*temp).is_reference() || exit_state.may_return()
         };
+        let live_srcs_in_dests = live_var.live_src_temps_in_dests(instr);
         match instr {
             Assign(_, _, src, AssignKind::Inferred) => {
                 if temp_needs_copy(src) {
```

Comment:
You can actually have identity assign (`$t1 = $t1`), so we need to do this here too.

Reply:
done